### PR TITLE
Fix: Assigned team(s) lost after invitation accept

### LINKED ISSUE #5759: Assigned team(s) lost after invitation accept
If I invite a member into my sentry installation and check a team in invitation screen the assignment is lost after the member registers on my server. I have to tick the team box again for every invited member.

### PR DESCRIPTION
**Pull request by nimishsinghal**
This closes #5759

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes require @getsentry/security sign-off</td>
    </tr>
  </tbody>
</table>

### Security concerns found

- src/sentry/web/frontend/auth_organization_login.py


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/web/frontend/auth_organization_login.py [hunk 1]
```diff
@@ -44,11 +44,14 @@ def handle_basic_auth(self, request, organization):
             user = register_form.save()
             user.send_confirm_emails(is_new_user=True)
 
-            defaults = {
-                'role': 'member',
-            }
+            try:
+                OrganizationMember.objects.get(email=user.email)
```
Comment (mattrobenolt) [inline src/sentry/web/frontend/auth_organization_login.py]: This query is most likely going to be wrong. I think we just want to check if this user is already a member of the organization they are joining at a minimum. In theory, you just want:

```python
if not OrganizationMember.objects.filter(user=user, organization=organization).exists():
    ....
```

But I'll need to confirm exactly.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

File: src/sentry/web/frontend/auth_organization_login.py (hunk 1 above)
Comment (nimishsinghal) [reply src/sentry/web/frontend/auth_organization_login.py]: @mattrobenolt do we need an organization check? Shouldn't we map the user if he is part of any organizations?

Comment (mattrobenolt) [reply src/sentry/web/frontend/auth_organization_login.py]: I'll have to look more closely to see exactly what's going on here. I'm not too intimately familiar, it just seems wrong at a glance. :)

**Comment by dcramer** [comment]
Heads up I'm going to invalidate this PR and fix the issue at the same time

(we're removing the automatic membership component for single org installs)

**Comment by dcramer** [comment]
I've removed the automatic membership in cf102c7c33d9aaf01346ee900d3c16750a6c9d85 (and unfortunately did not update the commit message when I rebased)

**Comment by dcramer** [comment]
Broke a test in prev commit, so reverted/recommitted with a proper message in a86d8ba4b

**Pull request closed without merging**